docs/content/docs/build/container-states-events.md

```diff
@@ -0,0 +1,336 @@
+
+Users, or your code, can start editing the data in the container right away, but the data is not shareable with other clients until the [container is published](./containers.md#publishing-a-container).
+Use the unpublished state to create initial data to populate your shared objects if needed.
+This is often useful in scenarios where you want to make sure that all connected clients have a coherent initial state.
+For example, if collaboration involves tables, your code can set a minimum table size.
+
+#### Publishing
+
+A container transitions to **publishing** state when `container.attach` method is called on the creating client. For details, see [Publishing a container](./containers.md#publishing-a-container). It is normally in this state only very briefly and then automatically transitions to **published** state when publication is complete.
+
+{{< callout note >}}
+
+In the Fluid APIs, the terms "attach", "attaching", "attached" and "detached" mean publish, publishing, published, and unpublished, respectively.
+
+{{< /callout >}}
+
+#### Published to the service
+
+When publishing completes, the container is **published**, so it exists in the Fluid service as well as the creating client. 
+Subsequent clients can [connect to the container](./containers.md#connecting-to-a-container), synchronize with its latest data values, and edit them.
+
+A container can never transition back to an unpublished state, but it can transition to a deleted state. 
+
+#### Deleted from the service
+
+A container transitions to **deleted** state when it is deleted. 
+Any connected clients become disconnected (although they may have been already disconnected before the deletion; see [Deleting a container from the service](#deleting-a-container-from-the-service)), but the container is not disposed on the clients. 
+Clients could continue to edit the container's data, but the data isn't shared anymore. Each client's changes are known only to it. 
+
+##### Deleting a container from the service
+
+Deleting a container is a service-specific feature, so you should refer to the documentation associated with the specific Fluid service you are using for instructions on how to do it. 
+To prevent clients from being suddenly cut off from data sharing, the service may put restrictions on when a container can be deleted. 
+For example, it may allow deletion only if all clients have been disconnected from the container for a specified period of time. 
+
+Once it is deleted, the container cannot move back to the **published** state again.
+It is possible for a client to create a new container using the same schema as a deleted container. 
+If a record of the final data values of the deleted container are available to the creating client, they can be used to initialize the new container. 
+But the new container has a different ID from the deleted one and subsequent clients will have to connect to it using the new ID.
```

Comment:
@alexvy86 We refer to the ID returned from `container.attach()` and the one passed to `getContainer()` as the container ID all over our docs, including the Getting Started docs. I suggest that we continue this because introducing the concept of "document" would be confusing. When the ID of the local `container` object is what's being referred to, let's call it "local container object ID" and, where helpful, not parenthetically that it is not the same as the container ID. How does that sound to you?

Reply:
I think that's alright for now to avoid over-complicating the docs but would still look for a place to put a callout note close to the "start" of the docs (not necessarily the start of this file, although it might be a good place for it too), to point out the fact that the terminology used here does not necessarily match one-to-one what they'd see when debugging/troubleshooting.

Reply:
@alexvy86 
I just noticed that our reference docs for the `FluidContainer` class and the `IFluidContainer` interface indicate that they do not have an `id` property. So, what is this "containerId" you are referring to that is distinct from the "documentId"?

Reply:
Ah... ok, we probably want to figure out which part of our API these docs are meant for. `IFluidContainer/FluidContainer` are the main entry points for users using the Azure client (probably most 3rd party devs), but they abstract away many things, so maybe the feedback I've been giving doesn't apply in that case. @ssimic2 is probably more familiar with that area. Sasha, as an initial question, how strong is the distinction between containerId and documentId in the Azure client API? Seems like it might be completely hidden from the users?